<?xml version="1.0"?>
<robot name="GLaDOS">

  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.383 0.38 0.02"/>
      </geometry>
      <origin rpy="0 0 3.141592654" xyz="-0.1185 0 0"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>

    <collision>
      <geometry>
        <box size="0.383 0.38 0.02"/>
      </geometry>
      <origin rpy="0 0 3.141592654" xyz="-0.1185 0 0"/>
    </collision>
  </link>

  <!-- Wheels -->

  <link name="left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.02" radius="0.0625"/>
      </geometry>
      <origin rpy="1.57075 0 0"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>

    <collision>
      <geometry>
        <cylinder length="0.02" radius="0.0625"/>
      </geometry>
      <origin rpy="1.57075 0 0"/>
    </collision>
  </link>

  <link name="right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.02" radius="0.0625"/>
      </geometry>
      <origin rpy="1.57075 0 0"/>
      <material name="blue"/>
    </visual>

    <collision>
      <geometry>
        <cylinder length="0.02" radius="0.0625"/>
      </geometry>
      <origin rpy="1.57075 0 0"/>
    </collision>
  </link>

  <joint name="base_to_lwheel" type="continuous">
    <parent link="base_link"/>
    <child link="left_wheel"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0 0.215 -0.0525"/>
  </joint>

  <joint name="base_to_rwheel" type="continuous">
    <parent link="base_link"/>
    <child link="right_wheel"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0 -0.215 -0.0525"/>
  </joint>

  <!-- Arm -->
  <link name="shoulder">
    <visual>
      <geometry>
        <box size="0.02 0.02 0.244"/>
      </geometry>
      <origin xyz="0.122 0 0" rpy="0 1.570796327 0"/>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
  </link>

  <joint name="base_to_shoulder" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder"/>
    <origin xyz="0.1 0 0.13"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="1.6" upper="-1.6" velocity="1.0"/>
  </joint>

  <link name="wrist">
    <visual>
      <geometry>
        <box size="0.065 0.02 0.02"/>
      </geometry>
      <origin xyz="0.0325 0 0" rpy="0 0 0"/>
      <material name="black"/>
    </visual>
  </link>

  <joint name="shoulder_to_wrist" type="revolute">
    <parent link="shoulder"/>
    <child link="wrist"/>
    <origin xyz="0.244 0 0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="1.6" upper="-1.6" velocity="1.0"/>
  </joint>

  <link name="gripper">
    <visual>
      <geometry>
        <box size="0.12 0.02 0.02"/>
      </geometry>
      <origin xyz="0.06 0 0" rpy="0 0 0"/>
      <material name="black"/>
    </visual>
  </link>

  <joint name="wrist_to_gripper" type="revolute">
    <parent link="wrist"/>
    <child link="gripper"/>
    <origin xyz="0.065 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="0" upper="-1.6" velocity="1.0"/>
  </joint>

  <link name="camera_base">
    <visual>
      <geometry>
        <box size="0.1 0.3 0.03"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

  <link name="camera_link"/>

  <joint name="base_to_camera" type="fixed">
    <parent link="base_link"/>
    <child link="camera_base"/>
    <origin xyz="-0.1143 -0.1778 0.4" rpy="0 0.698131701 0"/>
  </joint>

  <joint name="camera_base_to_camera_link" type="fixed">
    <parent link="camera_base"/>
    <child link="camera_link"/>
    <origin xyz="-0.018 0 0"/>
  </joint>

</robot>
